lib/modules/manager/vendir/extract.spec.ts

```diff
@@ -0,0 +1,118 @@
+import { Fixtures } from '../../../../test/fixtures';
+import { extractPackageFile } from '.';
+
+const oneContents = Fixtures.get('one-contents.yaml');
+const ociContents = Fixtures.get('oci-contents.yaml');
+const aliasContents = Fixtures.get('alias-contents.yaml');
+const multipleContents = Fixtures.get('multiple-contents.yaml');
+const nonHelmChartContents = Fixtures.get('non-helmchart.yaml');
+const emptyDirectories = Fixtures.get('empty-directories.yaml');
+
+describe('modules/manager/vendir/extract', () => {
+  describe('extractPackageFile()', () => {
+    it('returns null for invalid yaml file content', () => {
+      const result = extractPackageFile('nothing here: [', 'vendir.yml', {});
+      expect(result).toBeNull();
+    });
+
+    it('returns null for empty yaml file content', () => {
+      const result = extractPackageFile('', 'vendir.yml', {});
+      expect(result).toBeNull();
+    });
+
+    it('returns null for empty directories key', () => {
+      const result = extractPackageFile(emptyDirectories, 'vendir.yml', {});
+      expect(result).toBeNull();
+    });
+
+    it('returns null for nonHelmChart key', () => {
+      const result = extractPackageFile(nonHelmChartContents, 'vendir.yml', {});
+      expect(result).toBeNull();
+    });
+
+    it('single chart - extracts helm-chart from vendir.yml correctly', () => {
+      const result = extractPackageFile(oneContents, 'vendir.yml', {});
+      expect(result).toMatchObject({
+        deps: [
+          {
+            currentValue: '7.10.1',
+            depName: 'contour',
+            datasource: 'helm',
+            registryUrls: ['https://charts.bitnami.com/bitnami'],
+          },
+        ],
+      });
+    });
+
+    it('single chart - extracts oci helm-chart from vendir.yml correctly', () => {
+      const result = extractPackageFile(ociContents, 'vendir.yml', {});
+      expect(result).toMatchObject({
+        deps: [
+          {
+            currentValue: '7.10.1',
+            depName: 'contour',
```

Comment:
```suggestion
            depName: 'contour',
            packageName: 'charts.bitnami.com/bitnami/contour',
```
this should still be valid